datastore/query_test.go

```diff
@@ -456,6 +459,86 @@ func TestFilterParser(t *testing.T) {
 	}
 }
 
+func TestFilterWithArgs(t *testing.T) {
+	testCases := []struct {
```

Comment:
Can we declare this separately so we can reuse between the two tests? Or test both `Filter` functions in the same test?

Reply:
We could, but we'd need to break it up into multiple smaller test cases first (since some should fail in `TestFilter()` but pass in `TestFilterField()`) or add multiple test case fields.

Reply:
That makes sense. The downside right now is the list is really long and hard to compare between the two. Perhaps `TestFilterField` tests all of the `TestFilter` cases, but not the other way around? Doesn't seem too onerous. Fine to have more than one group of tests.

Reply:
Eh, you're right. It's better to have fewer test cases to maintain.